Producing a list of multiple help entries with hyperlinks

olexandr-konovalov opened the pull request:
This now improves help usability by displaying hyperlinks as shown on the picture.
![screen shot 2016-02-12 at 23 59 23](https://cloud.githubusercontent.com/assets/5751387/13023849/ec65e9bc-d1e4-11e5-8c5b-1b69a39173f7.png)

Commit:
Producing a list of multiple help entries with hyperlinks

alex-konovalov commented:
Interesting side-effect: if now call
```
SetHelpViewer("browser");
```
then entering ? or ?? results in opening all matches, each in a separate tab. That's not something what we want - it should produce the same picture as above, with links pointing to local files.

olexandr-konovalov commented on wrapper-kernel/jupyter_gap_wrapper/gap/setup.g:
```diff
@@ -66,10 +66,58 @@ HELP_VIEWER_INFO.jupyter_online :=
+    data := HELP_BOOK_HANDLER.(book.handler).HelpData(book, entrynr, type);
+    if data <> fail then
+      # show the data
+      HELP_VIEWER_INFO.(viewer).show(data);
```
It is this line which calls `HELP_VIEWER_INFO.browser` and causes opening of all tabs. I suggest actually to not to use `HELP_VIEWER_INFO.browser` at all. We should have `HELP_VIEWER_INFO.jupyter_online` and its counterpart `HELP_VIEWER_INFO.jupyter_local` which does not translate URLs to point to the server hosting the notebook.

alex-konovalov commented:
On a 2nd thought, having `HELP_VIEWER_INFO.jupyter_online` and its counterpart `HELP_VIEWER_INFO.jupyter_local` puts an unreasonable pressure on the user to decide which is used. It may be better to use `HELP_VIEWER_INFO.jupyter` (with a shorter name) and then it should convert URL or leave it local dependently on some parameter we may store in GAPInfo.

alex-konovalov commented:
@markuspf oops - I've tried to create `HELP_VIEWER_INFO.jupyter_local` as a quick way to make locally installed help usable, but then Jupyter prepends path to local file by `http://localhost:8888/`. If I will insert `file:///` then the URL is correct, but Jupyter does not allow to open it by clicking - only when manually copying URL and pasting it in another tab. Any ideas?

Commit:
Added HELP_VIEWER_INFO.jupyter_local help viewer

It works provided the jupyter notebook is started in the GAP root directory, since it should be able to serve the files under http://localhost:8888/files/doc/...

alex-konovalov commented:
I have rebased my PR on the current master and started to work on a version of `HELP` returning JSON record instead of printing the output. I will push new version as soon as I will have it usable.

Commit:
Modified HELP and related changes for nicer hyperlinks

alex-konovalov commented:
PR updated - now shows much nicer and usable output!

![screen shot 2016-02-16 at 13 18 06](https://cloud.githubusercontent.com/assets/5751387/13077310/e2fbee6c-d4af-11e5-8de4-cd26a0a20050.png)
